compiler/src/dotty/tools/dotc/util/Signatures.scala

```diff
@@ -41,109 +43,238 @@ object Signatures {
+   *
+   * @param pos Position for which call should be returned
+   * @return A triple containing the index of the parameter being edited, the index of functeon
+   *         being called, the list of overloads of this function).
+   */
+  def signatureHelp(pos: SourcePosition)(using Context): (Int, Int, List[Signature]) = {
+    val path = Interactive.pathTo(ctx.compilationUnit.tpdTree, pos.span)
+    computeSignatureHelp(path, pos.span)(using Interactive.contextOfPath(path))
+  }
+
+  /**
+   * Computes call info (current parameter index, function index, functions) for a method call.
    *
    * @param path The path to the function application
    * @param span The position of the cursor
    * @return A triple containing the index of the parameter being edited, the index of the function
    *         being called, the list of overloads of this function).
    */
-  def callInfo(path: List[tpd.Tree], span: Span)(using Context): (Int, Int, List[SingleDenotation]) =
-    val enclosingApply = path.dropWhile {
-      case apply @ Apply(fun, _) => fun.span.contains(span) || apply.span.end == span.end
-      case unapply @ UnApply(fun, _, _) => fun.span.contains(span) || unapply.span.end == span.end || isTuple(unapply)
-      case _ => true
-    }.headOption
-
-    enclosingApply.map {
+  def computeSignatureHelp(path: List[tpd.Tree], span: Span)(using Context): (Int, Int, List[Signature]) =
+    findEnclosingApply(path, span) match
+      case tpd.EmptyTree => (0, 0, Nil)
+      case Apply(fun, params) => applyCallInfo(span, params, fun)
       case UnApply(fun, _, patterns) => unapplyCallInfo(span, fun, patterns)
-      case Apply(fun, params) => callInfo(span, params, fun, Signatures.countParams(fun))
-    }.getOrElse((0, 0, Nil))
 
-  def callInfo(
-    span: Span,
-    params: List[tpd.Tree],
-    fun: tpd.Tree,
-    alreadyAppliedCount : Int
-  )(using Context): (Int, Int, List[SingleDenotation]) =
-    val paramIndex = params.indexWhere(_.span.contains(span)) match {
-      case -1 => (params.length - 1 max 0) + alreadyAppliedCount
-      case n => n + alreadyAppliedCount
+  /**
+   * Finds enclosing application from given `path` for `span`.
+   *
+   * @param path The path to the function application
+   * @param span The position of the cursor
+   * @return Tree which encloses closest application containing span.
+   *         In case if cursor is pointing on closing parenthesis and
+   *         next subsequent application exists, it returns the latter
+   */
+  private def findEnclosingApply(path: List[tpd.Tree], span: Span)(using Context): tpd.Tree =
+    path.filterNot {
+      case apply @ Apply(fun, _) => fun.span.contains(span) || isTuple(apply)
+      case unapply @ UnApply(fun, _, _) => fun.span.contains(span) || isTuple(unapply)
+      case _ => true
+    } match {
+      case Nil => tpd.EmptyTree
+      case direct :: enclosing :: _ if direct.source(span.end -1) == ')' => enclosing
```

Comment:
It handles apply signature at closing parenthesis. We should show second enclosing apply only when it exists and first one is already closed. Fixes:
```scala
object a {
  List(1, 2@@
}
```
Previous implementation returned nothing